client/src/pages/Results/index.tsx

```diff
@@ -86,66 +68,21 @@ const ResultsPage = () => {
       ),
     [results, page],
   );
-  useEffect(() => {
-    setSymbolsCollapsed(onlySymbolResults);
-  }, [onlySymbolResults]);
-
-  useEffect(() => {
-    if (!data) {
-      return;
-    }
-    const firstItem = data.data[0];
-    switch (firstItem?.kind) {
-      case 'dir':
-        setResults([]);
-        setOpenResult(null);
-        setDirResults(mapDirResult(firstItem));
-        break;
-      case 'file':
-        setOpenResult(mapFileResult(firstItem));
-        getHoverables(
-          firstItem.data.relative_path,
-          firstItem.data.repo_ref,
-        ).then((data) => {
-          setOpenResult((prevState) => ({
-            ...prevState!,
-            hoverableRanges: mapRanges(data.ranges),
-          }));
-        });
-        break;
-      default:
-        setDirResults(null);
-        setOpenResult(null);
-        setIsFileNav(false);
-        if (page === 0) {
-          setTotalPages(data.metadata.page_count);
-          setTotalCount(data.metadata.total_count);
-        }
-        setFilters(mapFiltersData(data.stats, filters));
-        setResults(mapResults(data));
-        break;
-    }
-
-    setInitialLoad(false);
-  }, [data]);
 
-  useEffect(() => {
-    if (!isFileNav) {
-      ref.current?.scrollTo({ top: 0, left: 0, behavior: 'smooth' });
+  const onResultClick = useCallback((repo: string, path?: string) => {
+    if (path) {
+      fileModalSearchQuery(`open:true repo:${repo} path:${path}`);
```

Comment:
shouldn't we also get hoverable ranges for this file?

Reply:
fixed